File: Sem4/MPP/cs/lab9server/Server/BiletService.cs
using Domain;
using log4net;
using Repository;

namespace WinFormsApp1.Service;

public class BiletService
{
    private static ILog logger = LogManager.GetLogger("BiletService");
    private BiletRepo biletRepo;
    private SpectacolRepo spectacolRepo;

    public BiletService(BiletRepo biletRepo,SpectacolRepo spectacolRepo)
    {
        this.biletRepo = biletRepo;
        this.spectacolRepo = spectacolRepo;
        logger.Info("BiletService initialized");
    }

    public void addBilet(String name, long spectacol, int seats)
    {
        try
        {
            var s = spectacolRepo.GetOne(spectacol);
            if (s.AvailableSeats < seats)
                throw new ServiceException("Not enough seats");
            s.AvailableSeats -= seats;
            s.SoldSeats+= seats;
            spectacolRepo.Update(s);

            Bilet b = new Bilet(name, seats, s);
            biletRepo.Add(b);
            logger.Info("Bilet added");

        }
        catch (Exception e)
        {
            Console.WriteLine(e);
            throw new ServiceException(e.Message);
        }
    }

}

File: Sem4/MPP/java/lab9client/Server/src/main/java/ubb/scs/mpp/Server/BiletService.java
package ubb.scs.mpp.Server;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ubb.scs.mpp.Service.ServiceException;
import ubb.scs.mpp.model.Bilet;
import ubb.scs.mpp.model.Spectacol;
import ubb.scs.mpp.Repository.BiletRepository;
import ubb.scs.mpp.Repository.RepoException;
import ubb.scs.mpp.Repository.SpectacolRepository;

public class BiletService {
    private static final Logger logger= LogManager.getLogger(BiletService.class);
    private BiletRepository repo;
    private SpectacolRepository srepo;
    public BiletService(BiletRepository repo,SpectacolRepository srepo) {
        this.repo = repo;
        this.srepo = srepo;
        logger.info("BiletService created");
    }
    public void AddBilet(String buyer,Long spectacol,Integer seats) throws ServiceException {
        try {
            Spectacol s;
            s= srepo.findOne(spectacol);
            if(s.getAvailable_seats() < seats)
                throw new ServiceException("Not enough available seats");
            s.setAvailable_seats(s.getAvailable_seats() - seats);
            s.setSold_seats(s.getSold_seats()+seats);

            srepo.update(s);
            Bilet bilet = new Bilet(buyer,seats,s);
            repo.save(bilet);
            logger.info("Bilet added");
        }catch(RepoException e) {
            throw new ServiceException(e.getMessage());
        }
    }
}
